src/System.Device.Gpio/System/Device/Gpio/RaspberryBoardIdentification.cs

```diff
@@ -0,0 +1,337 @@
+            {
+                if (_settings.TryGetValue("Serial", out string serial)
+                    && !string.IsNullOrEmpty(serial))
+                {
+                    return serial;
+                }
+
+                return string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether board is overclocked.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if board is overclocked; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsOverclocked
+        {
+            get
+            {
+                var firmware = Firmware;
+                return (firmware & 0xFFFF0000) != 0;
+            }
+        }
+
+        #endregion
+
+        #region Private Helpers
+
+        private Processor LoadProcessor(Model model)
+        {
+            switch (model)
+            {
+                case Model.RaspberryPiA:
+                case Model.RaspberryPiAPlus:
+                case Model.RaspberryPiBRev1:
+                case Model.RaspberryPiBRev2:
+                case Model.RaspberryPiBPlus:
+                case Model.RaspberryPiComputeModule:
+                case Model.RaspberryPiZero:
+                case Model.RaspberryPiZeroW:
+                    return Processor.Bcm2708;
+                case Model.RaspberryPi2B:
+                // TBC: B3(+) should be a BCM2710 processor ...
+                case Model.RaspberryPi3B:
+                case Model.RaspberryPi3BPlus:
+                case Model.RaspberryPiComputeModule3:
+                    return Processor.Bcm2709;
+                case Model.RaspberryPi4:
+                    return Processor.Bcm2711;
+                default:
+                    return Processor.Unknown;
+            }
+        }
+
+        /// <summary>
+        /// Detect the board CPU information from /proc/cpuinfo
+        /// </summary>
+        /// <returns>
+        /// The <see cref="RaspberryBoardIdentification"/>.
+        /// </returns>
+        internal static RaspberryBoardIdentification LoadBoard()
+        {
+            try
+            {
+                const string filePath = "/proc/cpuinfo";
+
+                var cpuInfo = File.ReadAllLines(filePath);
+                var settings = new Dictionary<string, string>();
+                var suffix = string.Empty;
+
+                foreach (var line in cpuInfo)
+                {
+                    var separator = line.IndexOf(':');
+
+                    if (!string.IsNullOrWhiteSpace(line) && separator > 0)
+                    {
+                        var key = line.Substring(0, separator).Trim();
+                        var val = line.Substring(separator + 1).Trim();
+                        if (string.Equals(key, "processor", StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            suffix = "." + val;
+                        }
+
+                        settings.Add(key + suffix, val);
```

Comment:
I'd rather have us parse the names here - perhaps create a list of locals up front, initialize them with null/-1 or some "not initialized" value and assign them directly

Reply:
Yea, I would have done the same, but for now I just did minimal changes to the existing PR. I can update it, though.

Reply:
I'd prefer if we did it